From be45ad064b7afdbc1e4f36c5a04c27cc364900c2 Mon Sep 17 00:00:00 2001
From: Praveenkumar I <ipkumar@codeaurora.org>
Date: Tue, 7 Nov 2017 15:03:52 +0530
Subject: [PATCH 8/8] clk: qcom: ipq8074: change freq table for
 port5_tx_clk_src

Originally QCOM clock framework assumes that there will be
only one entry for each freq but in port5, the same freq can be
supplied by 2 sources, uniphy0 and uniphy1. We need to move
uniphy1 above uniphy0 in frequency table so that uniphy1 will be
selected instead of uniphy0 if uniphy0 is running in
125 Mhz and uniphy1 is running in 312 Mhz.

Signed-off-by: Abhishek Sahu <absahu@codeaurora.org>
(cherry picked from commit 559fcf09e9681f7d1fcbd981a5de7957ffb3e496)
Signed-off-by: Praveenkumar I <ipkumar@codeaurora.org>

Change-Id: I91c17714922afc1ef4cdb5e6e47e2e813e2e9777
---
 drivers/clk/qcom/gcc-ipq8074.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/clk/qcom/gcc-ipq8074.c
+++ b/drivers/clk/qcom/gcc-ipq8074.c
@@ -1851,11 +1851,11 @@ static struct clk_regmap_div nss_port5_r
 
 static const struct freq_tbl ftbl_nss_port5_tx_clk_src[] = {
 	F(19200000, P_XO, 1, 0, 0),
-	F(25000000, P_UNIPHY0_TX, 5, 0, 0),
 	F(25000000, P_UNIPHY1_TX, 12.5, 0, 0),
+	F(25000000, P_UNIPHY0_TX, 5, 0, 0),
 	F(78125000, P_UNIPHY1_TX, 4, 0, 0),
-	F(125000000, P_UNIPHY0_TX, 1, 0, 0),
 	F(125000000, P_UNIPHY1_TX, 2.5, 0, 0),
+	F(125000000, P_UNIPHY0_TX, 1, 0, 0),
 	F(156250000, P_UNIPHY1_TX, 2, 0, 0),
 	F(312500000, P_UNIPHY1_TX, 1, 0, 0),
 	{ }
